testutil/integration/network/config.go

```diff
@@ -0,0 +1,65 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+package network
+
+import (
+	testtx "github.com/evmos/evmos/v14/testutil/tx"
+	"github.com/evmos/evmos/v14/utils"
+
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Config defines the configuration for a chain.
+// It allows for customization of the network to adjust to
+// testing needs.
+type Config struct {
+	chainID            string
+	amountOfValidators int
+	preFundedAccounts  []sdktypes.AccAddress
+	denom              string
+}
+
+// DefaultConfig returns the default configuration for a chain.
+func DefaultConfig() Config {
+	account, _ := testtx.NewAccAddressAndKey()
+	return Config{
+		chainID:            utils.MainnetChainID + "-1",
+		amountOfValidators: 3,
+		// No funded accounts besides the validators by default
+		preFundedAccounts: []sdktypes.AccAddress{account},
+		denom:             utils.BaseDenom,
+	}
+}
+
+// ConfigOption defines a function that can modify the NetworkConfig.
+// The purpose of this is to force to be declarative when the default configuration
+// requires to be changed.
+type ConfigOption func(*Config)
+
+// WithChainID sets a custom chainID for the network.
+func WithChainID(chainID string) ConfigOption {
+	return func(cfg *Config) {
+		cfg.chainID = chainID
+	}
+}
```

Comment:
The idea behind this approach is for any changes to the cfg to be declarative at the moment of the network init instead of passing the cfg struct that can be modified in different parts. I actually ilke the approach as it will minimize the opportunity for users to produce unclear code